### PR TITLE
Documentation fixes

### PR DESCRIPTION
**Pull request by tiennou**
Note: 8149b17 removes git_remote_rename_problem_cb. Not sure when it stopped being used though.

### CODE REVIEW

File: include/git2/transaction.h [hunk 1]
```diff
@@ -8,6 +8,14 @@
 #define INCLUDE_git_transaction_h__
 
 #include "common.h"
+
+/**
+ * @file git2/transaction.h
+ * @brief Git transaction routines.
+ * @defgroup Git transaction routines.
```
Comment (pks-t) [inline include/git2/transaction.h]: This should be `@defgroup git_transaction Git transaction routines` instead. Please also remove the trailing dots from `@brief` and `@defgroup`.

**Comment by pks-t** [comment]
Oh, one more thing. Could you maybe clarify the commit messages a bit by prefixing them with the module they're touch primaily, please? E.g. "remote: fix indentation".

Thanks for your work :+1:

File: include/git2/transaction.h (hunk 1 above)
Comment (carlosmn) [inline include/git2/transaction.h]: We don't actually use these blocks for anything with our current documentation generator, so it should just go away.

File: include/git2/describe.h [hunk 2]
```diff
@@ -105,6 +105,9 @@ typedef struct {
 
 GIT_EXTERN(int) git_describe_init_format_options(git_describe_format_options *opts, unsigned int version);
 
+/**
+ * An opaque struct that stores the result of a describe operation.
+ */
 typedef struct git_describe_result git_describe_result;
```
Comment (carlosmn) [inline include/git2/describe.h]: The opaqueness is not something we need to specify in the text.

File: include/git2/transaction.h (hunk 1 above)
Comment (tiennou) [inline include/git2/transaction.h]: I was trying to make docurium document those functions. Wasn't sure if the group being defined was a prerequisite, but I'll take them out then.

File: include/git2/describe.h (hunk 2 above)
Comment (tiennou) [inline include/git2/describe.h]: Will fix.

**Comment by tiennou** [comment]
Patch series redone from scratch.

`docurium` is giving me a hard time tough, as some things are left out of the parsing (eg. the whole of `transaction.h`, and things like `git_branch_remote_name` don't appear in the docs). Using the debug-heavy version I have here, it seems that clang sometimes misinterprets comments in some cases (eg, I get lots of `:comment_null`).

**Pull request merged**